### DISCUSSION TITLE
Problem solving - generating types from resolver with arguments

### DISCUSSION THREAD

**Discussion by MincePie** [MichalLytek/type-graphql#1041] [Q&A]
I'm trying to follow Ben Awad's lireddit tutorial.

At 6.49 he shows how to add arguments to a resolver query, with types that replace default values (in this case, replacing a default Float value, with Int).

The query is:

```
@Resolver()
  export class PostResolver {
    @Query(() => [Post])
    async posts(
      @Arg("limit", () => Int) limit: number,
      @Arg("cursor", () => String, { nullable: true }) cursor: string | null
    ): Promise<Post[]> {
      const realLimit = Math.min(50, limit);
      
      const qb = getConnection()
        .getRepository(Post)
        .createQueryBuilder("p")
        .orderBy('"createdAt"', "DESC")
        .take(realLimit)
        
        if (cursor) {
          qb.where('"createdAt" < :cursor',  new Date ( parseInt(cursor)) )
        }
       return qb.getMany();
    }
```
  
When I try to generate types from this resolver, I get 3 errors, being:

GraphQLDocumentError: Unknown type "Int". [Note: Int is imported from type-graphql]
Unknown argument "cursor" on field "Query.posts".
Unknown argument "limit" on field "Query.posts".
I'm trying to find things I can do to try and solve this problem.

When I use the playground to inspect the documents, I don't get any arguments feedback.

This[ SO post](https://stackoverflow.com/questions/69261934/graphqldocumenterror-unknown-type-int) includes a screenshot that shows the feedback I can get from the graphql playground. It doesn't include limit or cursor arguments at all.

**Comment by MichalLytek** [comment]
>  Unknown type "Int". [Note: Int is imported from type-graphql]

On screenshot you see that the type of `id` is still Float. And your `posts` query does not report any arguments.

Do you use babel/aws lambda or something? Non-standard TS Node setup requires speciall effort for decorators and metadata.

**Reply by MincePie** [reply]
Hi Michael, no I'm not using aws lamda or babel. I'm not sure why I can't get the generator to run to make the types. Do you have any ideas of where I might look for clues about why this generator won't run.

**Reply by MichalLytek** [reply]
If that's your repo:
https://github.com/armandsalle/ben-awad-lireddit-server/blob/main/src/resolvers/post.ts
You have setup that refers `dist` folder, so when you make a change in `src` it won't affect the app.
Please use `ts-node` directly, without the complication step.

**Reply by MincePie** [reply]
No - it's not my repo. That's someone else's attempt at this, but I do have a dist folder. Thank you for the suggestion. I'll start googling how I might be able to do this project without a dist folder. Thanks very much for the suggestion Michal.

**Comment by MincePie** [comment]
I know this is off topic for this package, but when I watch this video explaining what ts-node can do, it still creates a dist folder with compiled js (my project package.json has ts-node installed already - but I think you might be suggesting that there is a way to use it so that the project doesn't separately compile js in a dist folder - and that might allow me to run my type generator script without the errors that are currently stopping me from interacting with graphql). 

Is there something you're thinking that can work without creating this folder? Thank you so much for your help.

**Reply by MichalLytek** [reply]
`ts-node` works in memory

**Reply by MincePie** [reply]
I'm not clever enough to make sense of that yet. Ill do some research on what ts-node might be able to do to prevent a dist folder from being generated and try to create the types again. Thank you for the suggestion.
